enhancements/implicit-catalog-versioning.md

```diff
@@ -0,0 +1,190 @@
+    replaces: etcdoperator.v0.9.0
+```
+
+The `replaces` field is the source for the update graph for each install channel in a given operator. When the `operator-registry` builds the update graph, it actually walks through each CSV in a given Operator and, for each channel, constructs a chain of CSVs that define the update path. Instead, we can just define our spec without the replaces field:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: etcdoperator.v0.9.2
+  namespace: placeholder
+spec:
+    displayName: etcd
+    description: Etcd Operator
+    version: 0.9.2
+```
+
+We would use the fact that 0.9.2 > 0.9.0 to implicitly assume that 0.9.0 is replaced by 0.9.2 through the rules of semantic version.
+
+To accomplish this:
+- Make `spec.version` a required field in the operator-registry project
+- Update the `operator-registry` to walk through the set of CSVs and construct the replacement chain by reading the `version` field instead of `replaces`
+- Add `version` as a key in the `operator-registry`'s bundle table to ensure that version has to be unique.
+- Add a `replacesVersion` field for each channel in the `operator-registry`'s database that defines which version this bundle is replacing in the update graph (along with a query to get the previous version)
+- Add a `replacedByVersion` field for each channel in the `operator-registry`'s database that defines which version this bundle is being replaced by in the update graph (along with a query to get the next version)
+- Update everywhere in OLM that references `spec.Replaces` to instead get the `replaceVersion` to determine what CSV the current is replacing *if the new fields are available*
+
+Note that these requirements will only be the case when generating update graphs with the latest version of the operator-registry's `opm registry` commands. For now, we will not update the version of the `ClusterServiceVersion` CRD type in OLM to allow for backwards compatibility
+
+### Skipping updates
+
+When OLM updates an operator,  The `spec.skips` field is used to bypass certain specific previous versions when building the update graph. This field uses the CSV `spec.name` to determine what specific CSVs to skip in the same way that the `replaces` field is used:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: etcdoperator.v0.9.2
+  namespace: placeholder
+  annotations:
+spec:
+    displayName: etcd
+    description: Etcd Operator
+    replaces: etcdoperator.v0.9.0
+    skips:
+    - etcdoperator.v0.9.1
+```
+
+There is *also* the `olm.skipRange` annotation, which if your operator already follows semver, can use semver range to skip multiple versions:
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+    name: elasticsearch-operator.v4.1.2
+    namespace: placeholder
+    annotations:
+        olm.skipRange: '>=4.1.0 <4.1.2'
```

Comment:
skiprange is pretty different from `skip`, because OLM queries the channel head when searching for updates and only respects skiprange on the channel head. my assumption would be that this is unnecessary depending on how we generate the graph for semver channels (i.e. because the latest z stream would explicitly skip all previous z-streams)

Reply:
They are different in implementation, but they essentially present the same purpose ("how can I define what versions to skip installing if a user is on some specific CSV?")

IMO the implicit nature of the versioning allows users some of that flexibility by themselves and in that case we just need one of these. Is there some useful edge case to discuss here?